CRM/Contribute/Form/ContributionBase.php

```diff
@@ -1544,11 +1544,25 @@ protected function assignTotalAmounts(): void {
    * @throws \CRM_Core_Exception
    */
   public function getCurrency(): string {
-    $currency = $this->getContributionPageValue('currency');
-    if (empty($currency)) {
-      $currency = CRM_Utils_Request::retrieveValue('currency', 'String');
+    if (!isset(\Civi::$statics[__CLASS__][__FUNCTION__]['currency'])) {
+      $existingContributionID = $this->getExistingContributionID();
+      if ($existingContributionID) {
+        $currency = \Civi\Api4\Contribution::get(FALSE)
+          ->addSelect('currency')
+          ->addWhere('id', '=', $existingContributionID)
```

Comment:
Since this function depends on contribution id, the statics key should depend on it too. Possibly in practice this function is only ever called for one id within a page run, even in tests, but this is civi, and people do all kinds of things.

Reply:
I've removed the caching